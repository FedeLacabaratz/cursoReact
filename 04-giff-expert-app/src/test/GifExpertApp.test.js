import React from 'react';
import '@testing-library/jest-dom';
import { shallow, ShallowWrapper } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('pruebas en <GifExpertApp />', () => {
    
    let wrapper = shallow(<GifExpertApp />);

    beforeAll(() => {
        jest.clearAllMocks();
        wrapper = shallow(<GifExpertApp />);
    })

    test('debe de imprimir correctamente el componente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['nirvana', 'van halen'];
        wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});
