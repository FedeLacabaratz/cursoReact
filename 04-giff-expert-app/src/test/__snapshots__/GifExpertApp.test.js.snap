// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe de imprimir correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="nirvana"
      key="nirvana"
    >
      nirvana
    </GifGrid>
    <GifGrid
      category="van halen"
      key="van halen"
    >
      van halen
    </GifGrid>
  </ol>
</Fragment>
`;
